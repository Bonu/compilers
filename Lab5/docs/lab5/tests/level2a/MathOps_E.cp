// Main: Test expressions of overloaded operators with mixed types.

class Main {

    string a = "";
    bool p = 0;

  // Assignments 
    string b = p;
    int i = a;
    bool q = "";
    int j = q;

  // Mathematical Expressions 
   // Binary Operators 

    int k = a * j;
    int l = p / j;		// int and bool 
    int m = j + a;		// int and string 
    int n = j - "5.0";

    int o = j * p;

    bool r = i;		// bool and int 
    bool s = b;		// bool and string 

   // Unary Operators 
    bool t = -p;
    bool u = +p;

}
