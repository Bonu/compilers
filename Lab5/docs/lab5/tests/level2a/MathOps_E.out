line 6: `p' and its initializing expression have incompatible types
          -- requires type `bool' but found ` 0 ' which has type `int'
line 9: `b' and its initializing expression have incompatible types
          -- requires type `string' but found ` p ' which has type `bool'
line 10: `i' and its initializing expression have incompatible types
          -- requires type `int' but found ` a ' which has type `string'
line 11: `q' and its initializing expression have incompatible types
          -- requires type `bool' but found ` "" ' which has type `string'
line 12: `j' and its initializing expression have incompatible types
          -- requires type `int' but found ` q ' which has type `bool'
line 17: invalid argument 1 ` a ' for arithmetic operator `*'
          -- expects a numeric argument but found string
line 18: invalid argument 1 ` p ' for arithmetic operator `/'
          -- expects a numeric argument but found bool
line 19: invalid argument 2 ` a ' for arithmetic operator `+'
          -- expects a numeric argument but found string
line 20: invalid argument 2 ` "5.0" ' for arithmetic operator `-'
          -- expects a numeric argument but found string
line 22: invalid argument 2 ` p ' for arithmetic operator `*'
          -- expects a numeric argument but found bool
line 24: `r' and its initializing expression have incompatible types
          -- requires type `bool' but found ` i ' which has type `int'
line 25: `s' and its initializing expression have incompatible types
          -- requires type `bool' but found ` b ' which has type `string'
line 28: invalid argument type for unary operator `-'
          -- expects a numeric argument but found bool
line 29: invalid argument type for unary operator `+'
          -- expects a numeric argument but found bool
line 0: Method `Main.main' has not been defined
15 errors were detected
